e3db-crypto-android/src/main/java/com/tozny/e3db/crypto/AndroidConfigStorageHelper.java

```
@@ -27,124 +27,155 @@
+            throw new IllegalArgumentException("Identifier string can only contain alphanumeric characters, underscores, and hyphens.");
 
-        if (identifier.length() > 127)
+        if (identifier.length() > 100) /* In case device file system limits filenames 127 characters (and we're adding roughly 25 characters). */
             throw new IllegalArgumentException("Identifier string cannot be more than 127 characters in length.");
 
         if (identifier.trim().length() == 0)
             throw new IllegalArgumentException("Identifier string cannot be empty.");
     }
 
-    private static void checkArgs(Context context, String identifier, String config) {
-        checkArgs(context, identifier);
+    private static void checkArgs(Context context, String identifier, String config, Object handler) {
+        checkArgs(context, identifier, handler);
 
         if (config == null)
             throw new IllegalArgumentException("Config string cannot be null.");
     }
 
-    public AndroidConfigStorageHelper(@NotNull Context context, @NotNull String identifier, KeyProtection protection, KeyAuthenticator keyAuthenticator) {
+    private final static String COM_TOZNY_E3DB_CRYPTO = "com.tozny.e3db.crypto-";
+
+    private static String full(String identifier, KeyProtection protection) {
+        if (protection == null)
+            return COM_TOZNY_E3DB_CRYPTO + identifier + "-NO";
+
+        switch (protection.protectionType()) {
+            case NONE:        return COM_TOZNY_E3DB_CRYPTO + identifier + "-NO";
+            case FINGERPRINT: return COM_TOZNY_E3DB_CRYPTO + identifier + "-FP";
+            case LOCK_SCREEN: return COM_TOZNY_E3DB_CRYPTO + identifier + "-LS";
+            case PASSWORD:    return COM_TOZNY_E3DB_CRYPTO + identifier + "-PW";
+        }
+
+        return COM_TOZNY_E3DB_CRYPTO + identifier + "-NO";
```

`switch` statement should instead have a `default` case that throws (fail early).